cmd/agent-actions/actions/delete_node_handler.go

```diff
@@ -3,27 +3,58 @@ package actions
 import (
 	"context"
 	"encoding/json"
-	"errors"
+	"time"
 
+	"github.com/cenkalti/backoff/v4"
+	"github.com/sirupsen/logrus"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 
 	"castai-agent/cmd/agent-actions/telemetry"
 )
 
-func newDeleteNodeHandler(clientset kubernetes.Interface) ActionHandler {
+type deleteNodeConfig struct {
+	deleteRetries   uint64
+	deleteRetryWait time.Duration
```

Comment:
In our API we have force delete flag. Shouldn't we bring it up to Node delation phase in k8s ?

Reply:
This action only deletes node from etcd, there is no need to any force. We need force only for node drain.